manifests/proxydns.pp

```diff
@@ -9,19 +9,32 @@
   # puppet fact names are converted from ethX.X and ethX:X to ethX_X
   # so for alias and vlan interfaces we have to modify the name accordingly
   $interface_fact_name = regsubst($foreman_proxy::dns_interface, '[.:]', '_')
-  $ip = inline_template("<%= scope.lookupvar('::ipaddress_${interface_fact_name}') %>")
+  $ip = getvar("::ipaddress_${interface_fact_name}")
 
   if ! is_ip_address($ip) {
     fail("Could not get the ip address from fact ipaddress_${interface_fact_name}")
   }
 
-  ::dns::zone { $foreman_proxy::dns_zone:
+  if $::foreman_proxy::dns_reverse {
+    $reverse = $::foreman_proxy::dns_reverse
+  } else {
+    $netmask = getvar("::netmask_${interface_fact_name}")
+    unless is_ip_address($netmask) {
+      fail("Could not get the netmask from fact netmask_${interface_fact_name}")
+    }
+    $reverse = get_network_in_addr($ip, $netmask)
+    if ! is_string($reverse) or $reverse == '' {
+      fail("Could not determine reverse for ${ip}/${netmask}")
+    }
+  }
+
+  ::dns::zone { $::foreman_proxy::dns_zone:
     soa     => $::fqdn,
     reverse => false,
     soaip   => $ip,
   }
 
-  ::dns::zone { $foreman_proxy::dns_reverse:
+  ::dns::zone { $reverse:
```

Comment:
This error?

Evaluation Error: Missing title. The title expression resulted in undef at /home/travis/build/theforeman/puppet-foreman_proxy/spec/fixtures/modules/foreman_proxy/manifests/proxydns.pp:37:17 on node testing-docker-24f9b60e-1948-419b-bfa3-add6f0487c67.ec2.internal

Maybe the if check above on  line 26 isn't working correctly, and reverse is getting passed as undef?